devcontainers/lib/dependabot/devcontainers/file_updater/config_updater.rb

```diff
@@ -0,0 +1,71 @@
+# typed: true
+# frozen_string_literal: true
+
+require "dependabot/file_updaters/base"
+require "dependabot/shared_helpers"
+require "dependabot/logger"
+require "dependabot/devcontainers/utils"
+
+module Dependabot
+  module Devcontainers
+    class FileUpdater < Dependabot::FileUpdaters::Base
+      class ConfigUpdater
+        def initialize(dependency:, config_dependency_file:, repo_contents_path:, credentials:)
+          @dependency = dependency
+          @config_dependency_file = config_dependency_file
+          @repo_contents_path = repo_contents_path
+          @credentials = credentials
+        end
+
+        def update
+          SharedHelpers.in_a_temporary_repo_directory(base_dir, repo_contents_path) do
+            File.write(config_name, config_contents)
+
+            SharedHelpers.with_git_configured(credentials: credentials) do
+              latest_major = dependency.requirements.first[:metadata][:latest_major]
+              try_config_update(dependency_name: dependency.name, target_version: latest_major)
+              return File.read(config_name), File.read(lockfile_name)
+            end
+          end
+        end
+
+        private
+
+        def base_dir
+          config_dependency_file.directory
+        end
+
+        def config_name
+          config_dependency_file.name
+        end
+
+        def lockfile_name
+          Utils.expected_lockfile_name(config_dependency_file.name)
+        end
+
+        def config_contents
+          config_dependency_file.content
+        end
+
+        def try_config_update(dependency_name:, target_version:)
+          cmd = "devcontainer upgrade"
+          args = "--workspace-folder . \
+          --feature #{dependency_name} \
+          --config #{config_name} \
+          --target-version #{target_version}"
+
+          puts "Running command: #{cmd}"
+          SharedHelpers.run_shell_command(
+            "#{cmd} #{args}",
+            stderr_to_stdout: false
+          )
+      rescue SharedHelpers::HelperSubprocessFailed => e
+        puts "!! [UPDATER] Failed to run #{cmd} command:\n#{e.message}"
+        Dependabot.logger.info("Failed to run #{cmd} command:\n#{e.message}")
+        end
```

Comment:
This is very weirdly indented, surprised rubocop did not catch it.

Also, it'd be better to not rescue this if it's not expected that it fails, so that we get issues reported in our internal tracker and don't leave bugs hidden in update job logs.